### PR TITLE
Pytorch schedulers

### PR DESCRIPTION
**Pull request by amitibo**
Description:

Add wrappers that allow using "native" `torch.optim lr_scheduler` objects as (or together with)`ignite.contrib.handlers.param_scheduler` objects. For example this allowing attaching `ReduceLROnPlateau` to a `trainer` object or concatenate it with other `ParamScheduler` objects.

Check list:
* [ ] New tests are added (if a new feature is modified)
* [x] New doc strings: text and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/contrib/handlers/param_scheduler.py [hunk 1]
```diff
@@ -284,3 +284,165 @@ def __call__(self, engine, name=None):
...
+            param_name='lr',
+            save_history=save_history
+        )
+
+    def __call__(self, engine, name=None):
+
+        #
+        # Call the pytorch scheduler step method.
+        #
+        self.lr_scheduler.step(engine.state.epoch)
```
Comment (vfdev-5) [inline ignite/contrib/handlers/param_scheduler.py]: I think we should not attach the scheduler to the epochs.
Why can't we simply do 
```python
self.lr_scheduler.last_epoch += 1
```
without any update by pytorch and update params from ParamScheduler side ?

File: ignite/contrib/handlers/param_scheduler.py [hunk 2]
```diff
@@ -284,3 +284,165 @@ def __call__(self, engine, name=None):
...
+        # Call the ignite object __call__ method.
+        # Note:
+        # The 'lr' parameter will be updated for the second time but with the same value.
+        #
+        super(LRScheduler, self).__call__(engine, name)
+
+    def get_param(self):
+        """Method to get current optimizer's parameter value
+        """
+        return self.lr_scheduler.get_lr()
```
Comment (vfdev-5) [inline ignite/contrib/handlers/param_scheduler.py]: This can be tricky, `lr_scheduler.get_lr()` return a list of values and not a single value.

**Comment by vfdev-5** [comment]
@amitibo I'm trying to check `LRScheduler` for some basic schedulers e.g. `StepLR`.
I left some comments in the code.

Should not we insure that the optimizer passed to an lr_scheduler should have a single param group ?
Let me explain:
```python
t1 = torch.zeros([1], requires_grad=True)
optimizer = torch.optim.SGD([t1], lr=0.1)
lr_scheduler = StepLR(optimizer=optimizer, step_size=2)
lr_scheduler.get_lr()
> [0.1]
```
and 
```python
t1 = torch.zeros([1], requires_grad=True)
t2 = torch.zeros([1], requires_grad=True)
t3 = torch.zeros([1], requires_grad=True)

optimizer = torch.optim.SGD(
    [
        {"params": t1, 'lr': 0.001},
        {"params": t2, 'lr': 0.01},
        {"params": t3, 'lr': 0.0001}
    ]
)
lr_scheduler = StepLR(optimizer=optimizer, step_size=2)
lr_scheduler.get_lr()
> [0.01, 0.1, 0.001]
```

Code of `lr_scheduler.step()` of `_LRScheduler` assign values as 
```python
for param_group, lr in zip(self.optimizer.param_groups, self.get_lr()):
            param_group['lr'] = lr
```
But in our case, 
```python
value = self.get_param()

for param_group in self.optimizer_param_groups:
    param_group[self.param_name] = value
```

**Comment by amitibo** [comment]
> @amitibo I'm trying to check `LRScheduler` for some basic schedulers e.g. `StepLR`.
> I left some comments in the code.
> 
> Should not we insure that the optimizer passed to an lr_scheduler should have a single param group ?
> Let me explain:
> 
> ```python
> t1 = torch.zeros([1], requires_grad=True)
> optimizer = torch.optim.SGD([t1], lr=0.1)
> lr_scheduler = StepLR(optimizer=optimizer, step_size=2)
> lr_scheduler.get_lr()
> > [0.1]
> ```
> 
> and
> 
> ```python
> t1 = torch.zeros([1], requires_grad=True)
> t2 = torch.zeros([1], requires_grad=True)
> t3 = torch.zeros([1], requires_grad=True)
> 
> optimizer = torch.optim.SGD(
>     [
>         {"params": t1, 'lr': 0.001},
>         {"params": t2, 'lr': 0.01},
>         {"params": t3, 'lr': 0.0001}
>     ]
> )
> lr_scheduler = StepLR(optimizer=optimizer, step_size=2)
> lr_scheduler.get_lr()
> > [0.01, 0.1, 0.001]
> ```
> 
> Code of `lr_scheduler.step()` of `_LRScheduler` assign values as
> 
> ```python
> for param_group, lr in zip(self.optimizer.param_groups, self.get_lr()):
>             param_group['lr'] = lr
> ```
> 
> But in our case,
> 
> ```python
> value = self.get_param()
> 
> for param_group in self.optimizer_param_groups:
>     param_group[self.param_name] = value
> ```

@vfdev-5, sorry for the late reply, I think it should be the other way round, `ParamScheduler` should support a per param group parameter. This way ignite can support the full capabilities of `pytroch.lr_scheduler` instead of downgrading it. I also think that it can be done without changing the current API.

**Comment by vfdev-5** [comment]
@amitibo could you please create an issue for that ?
